First we have to import the pandas library to allow us to read the csv file

In [23]:
import pandas as pd

Next we need to load the csv into pandas dataframe. Going to pull this from he GitHub Repository located at https://github.com/hadley/nycflights13/tree/master/data-raw. This will make it so anyone can run the code without needing to download the file and change paths.

In [35]:
airports = pd.read_csv("https://raw.githubusercontent.com/tidyverse/nycflights13/main/data-raw/airports.csv")
airports.head(5)

faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York

Now we are going to sort the data by latitude in order to help us answer question 1. 
What is the northernmost airport in the United States?

In [36]:
northernmost = airports.sort_values("lat", ascending=False)
northernmost.head(5)

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage

Based on my code the northernmost airport looks to be "Dillant Hopkins Airport" with a lat of 72.270833. I believe this to an error as this airport is located in New Hampshire. I double checked the airport and it looks like the lat and lon were swapped.

This would mean the real northernmost airport is Wiley Post Will Rogers Mem at 71.285446. This makes more sense as this airport is located in Alaska 

I dowloaded the CSV and updated it with the values swapped and uploaded it to my GitHub repository, so I can run the code with the correct values and condensed into 1 cell

In [39]:
airports = pd.read_csv("https://raw.githubusercontent.com/KcirtapdiaZ/IS_362_Data_Acquisition_and_Mgmt/main/airports.csv")
airports.head(5)
northernmost = airports.sort_values("lat", ascending=False)
northernmost.head(5)

faa                                        name        lat         lon  \
230   BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110   AIN                          Wainwright Airport  70.638056 -159.994722   
708   K03                               Wainwright As  70.613378 -159.860350   
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   
1363  UUK                        Ugnu-Kuparuk Airport  70.330833 -149.597500   

      alt  tz dst              tzone  
230    44  -9   A  America/Anchorage  
110    41  -9   A  America/Anchorage  
708    35  -9   A  America/Anchorage  
152    96  -9   A  America/Anchorage  
1363   67  -9   A  America/Anchorage

Next we have to answer question 2.
What is the easternmost airport in the United States?
In order to find the easternmost airport we can use a similar code to sort the values with lon instead of lat

In [40]:
easternmost = airports.sort_values("lon", ascending=False)
easternmost.head(5)

faa                           name        lat         lon   alt  tz dst  \
1290  SYA                   Eareckson As  52.712275  174.113620    98  -9   A   
942   MYF               Montgomery Field  32.475900  117.759000    17   8   A   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
417   EEN        Dillant Hopkins Airport  42.898333   72.270833   149  -5   A   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   

                  tzone  
1290  America/Anchorage  
942      Asia/Chongqing  
396      Asia/Chongqing  
417                 NaN  
444    America/New_York

Looking at the data chart it looks like Eareckson As is the easternmost airport with a lon of 174.113620

For question 3 we need to pull in a different csv file. I will pull it from the same repository as before to make sure the code can be run without needing to save the file locally

In [114]:
weather = pd.read_csv("https://raw.githubusercontent.com/tidyverse/nycflights13/main/data-raw/weather.csv")
weather.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  wind_speed  \
0    EWR  2013      1    1     1  39.02  26.06  59.37     270.0    10.35702   
1    EWR  2013      1    1     2  39.02  26.96  61.63     250.0     8.05546   
2    EWR  2013      1    1     3  39.02  28.04  64.43     240.0    11.50780   
3    EWR  2013      1    1     4  39.92  28.04  62.21     250.0    12.65858   
4    EWR  2013      1    1     5  39.02  28.04  64.43     260.0    12.65858   

   wind_gust  precip  pressure  visib             time_hour  
0        NaN     0.0    1012.0   10.0  2013-01-01T06:00:00Z  
1        NaN     0.0    1012.3   10.0  2013-01-01T07:00:00Z  
2        NaN     0.0    1012.5   10.0  2013-01-01T08:00:00Z  
3        NaN     0.0    1012.2   10.0  2013-01-01T09:00:00Z  
4        NaN     0.0    1011.9   10.0  2013-01-01T10:00:00Z

Question 3 is On February 12th, 2013, which New York area airport had the windiest weather?
In order to find this we will need to do some filtering.

In [115]:
feb122013 = weather[(weather["month"] == 2) & (weather["day"] == 12) & (weather["year"] == 2013)]
feb122013.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1006    EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1007    EWR  2013      2   12     1  39.92  37.94  92.56     250.0   
1008    EWR  2013      2   12     2  39.92  28.04  62.21     270.0   
1009    EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
1010    EWR  2013      2   12     4  39.02  26.96  64.29     280.0   

      wind_speed  wind_gust  precip  pressure  visib             time_hour  
1006     6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1007     9.20624        NaN     0.0    1007.2   10.0  2013-02-12T06:00:00Z  
1008    20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z  
1009  1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
1010    12.65858        NaN     0.0    1008.8   10.0  2013-02-12T09:00:00Z

We are looking for the airport with the windiest weather so we will need to sort by the wind speed column.

In [116]:
feb122013.sort_values('wind_speed', ascending=False,)
feb122013.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1006    EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1007    EWR  2013      2   12     1  39.92  37.94  92.56     250.0   
1008    EWR  2013      2   12     2  39.92  28.04  62.21     270.0   
1009    EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
1010    EWR  2013      2   12     4  39.02  26.96  64.29     280.0   

      wind_speed  wind_gust  precip  pressure  visib             time_hour  
1006     6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1007     9.20624        NaN     0.0    1007.2   10.0  2013-02-12T06:00:00Z  
1008    20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z  
1009  1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
1010    12.65858        NaN     0.0    1008.8   10.0  2013-02-12T09:00:00Z

Now we can see a value with a wind speed over 1000 which is definately an error. This needs to be removed. I am going to use the < function to ensure I dont pull wind speeds over 100.

I will create a new variable to hold the new data after using the < function

In [117]:
feb122013_clean = feb122013[feb122013['wind_speed'] < 200]
feb122013_clean.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1006    EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1007    EWR  2013      2   12     1  39.92  37.94  92.56     250.0   
1008    EWR  2013      2   12     2  39.92  28.04  62.21     270.0   
1010    EWR  2013      2   12     4  39.02  26.96  64.29     280.0   
1011    EWR  2013      2   12     5  39.92  26.96  60.29     270.0   

      wind_speed  wind_gust  precip  pressure  visib             time_hour  
1006     6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1007     9.20624        NaN     0.0    1007.2   10.0  2013-02-12T06:00:00Z  
1008    20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z  
1010    12.65858        NaN     0.0    1008.8   10.0  2013-02-12T09:00:00Z  
1011    13.80936        NaN     0.0       NaN   10.0  2013-02-12T10:00:00Z

I could've added the sorting with the new variable to the above code but I figured I'd seperate it so easier visibility. I needed to add the .head method to the end of this line of code otherwise the data would not sort properly again.

In [119]:
feb122013_clean.sort_values(by='wind_speed', ascending=False).head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
9717     JFK  2013      2   12     8  42.08  24.08  48.56     280.0   
1008     EWR  2013      2   12     2  39.92  28.04  62.21     270.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
9717     20.71404   27.61872     0.0    1011.6   10.0  2013-02-12T13:00:00Z  
1008     20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z

Looking at the sorted data chart we can see that LGA had the windiest weather on February 12th 2013